sessionctx/binloginfo/binloginfo.go

```diff
@@ -114,11 +119,11 @@ func (info *BinlogInfo) WriteBinlog(clusterID uint64) error {
 	}
 
 	if err != nil {
-		cfg := config.GetGlobalConfig()
-		if cfg.Binlog.IgnoreError {
+		if atomic.LoadUint32(&ignoreError) == 1 {
 			log.Errorf("critical error, write binlog fail but error ignored: %s", errors.ErrorStack(err))
 			metrics.CriticalErrorCounter.Add(1)
-			atomic.StoreUint32(&errorStopped, 1)
+			// If error happens once, we'll stop writing binlog.
+			atomic.CompareAndSwapUint32(&stopped, valueStopped, valueStopped+1)
```

Comment:
How about just `atomic.StoreUint32(1)`?

Reply:
I write this code deliberately. @coocood 

Consider this scenario:

```
txn1 ... WriteBinlog() ...  check stopped = 0 ... continue
txn2 ... WriteBinlog() ...  check stopped = 0 ... continue
....
txn2 ... run here ... atomic.StoreUint32(&stopped, 1) ... so stopped is 1

http.ResetStopFlag()     // After this, we all think stop flag has been reset... but ...

txn1 ... run here ... atomic.StoreUint32(&stopped, 1) .... 

What the fuck? ResetStopFlag operation is gone,  we lost update!
```